ValueChangedGenerator/ValueChangedGenerator/ValueChangedGenerator.Test/ValueChangedGeneratorUnitTests.cs

```diff
@@ -55,5 +58,214 @@ class TYPENAME
+                }
+
+                set
+                {
+                    SetProperty(ref _value.Y, value, YProperty);
+                    OnPropertyChanged(ZProperty);
+                }
+            }
+
+            private static readonly PropertyChangedEventArgs YProperty = new PropertyChangedEventArgs(nameof(Y));
+            /// <summary>
+            /// Name.
+            /// </summary>
+            public string Name
+            {
+                get
+                {
+                    return _value.Name;
+                }
+
+                set
+                {
+                    SetProperty(ref _value.Name, value, NameProperty);
+                }
+            }
+
+            private static readonly PropertyChangedEventArgs NameProperty = new PropertyChangedEventArgs(nameof(Name));
+            public int Z => _value.Z;
+            private static readonly PropertyChangedEventArgs ZProperty = new PropertyChangedEventArgs(nameof(Z));
+        }
+    }
+}";
+            var text = @"
+public partial class C1 : BindableBase
+{
+    struct NotifyRecord
+    {
+        public int X;
+        public int Y;
+        public int Z => X * Y;
+
+        /// <summary>
+        /// Name.
+        /// </summary>
+        public string Name;
+    }
+
+    public partial class C2 : BindableBase
+    {
+        struct NotifyRecord
+        {
+            public int X;
+            public int Y;
+            public int Z => X * Y;
+
+            /// <summary>
+            /// Name.
+            /// </summary>
+            public string Name;
+        }
+
+        public partial class C3 : BindableBase
+        {
+            struct NotifyRecord
+            {
+                public int X;
+                public int Y;
+                public int Z => X * Y;
+
+                /// <summary>
+                /// Name.
+                /// </summary>
+                public string Name;
+            }
+        }
+    }
+}";
+            var compilation = CSharpCompilation.Create(assemblyName: "test", options: new CSharpCompilationOptions(OutputKind.DynamicallyLinkedLibrary))
+                .AddSyntaxTrees(CSharpSyntaxTree.ParseText(text));
+            var generator = new Generator();
+            var syntaxTrees = compilation.SyntaxTrees;
+            Assert.Single(syntaxTrees);
+            var classDecl = (ClassDeclarationSyntax)syntaxTrees.First().GetCompilationUnitRoot()
+                .DescendantNodes().First(x => x is ClassDeclarationSyntax)
+                .DescendantNodes().First(x => x is ClassDeclarationSyntax)
+                .DescendantNodes().First(x => x is ClassDeclarationSyntax);
+            var model = compilation.GetSemanticModel(classDecl.SyntaxTree);
+            var container = model.GetDeclaredSymbol(classDecl);
+            Assert.NotNull(container);
+            Assert.Equal(expected, actual: generator.GeneratePartialDeclaration(container!, classDecl).ToFullString());
+        }
```

Comment:
I'm really not sure if this is the right way to write tests, but I tried my best.

Reply:
I'm also not sure. Source Generators are not so mature. This code looks good so far.